plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3select/TestS3SelectPushdown.java

```diff
@@ -43,4 +94,93 @@ public void testIsCompressionCodecSupported()
         assertFalse(S3SelectPushdown.isCompressionCodecSupported(inputFormat, new Path("s3://fakeBucket/fakeObject.snappy")));
         assertTrue(S3SelectPushdown.isCompressionCodecSupported(inputFormat, new Path("s3://fakeBucket/fakeObject.bz2")));
     }
+
+    @Test
+    public void testShouldEnableSelectPushdown()
```

Comment:
Nice to see more tests added to this class. Can we add some tests around when scan range should be enabled/disabled?

Reply:
Sounds good, I'll extract the checks in a static method and add tests for it.

Reply:
Updated, please check